plonk-napi/src/plonk_verifier_index/fq.rs

```diff
@@ -96,20 +213,138 @@ pub fn caml_pasta_fq_plonk_verifier_index_shifts(
     let s = shifts.shifts();
 
     Ok(NapiFqShifts {
-        s0: serialize_fp(&s[0])?,
-        s1: serialize_fp(&s[1])?,
-        s2: serialize_fp(&s[2])?,
-        s3: serialize_fp(&s[3])?,
-        s4: serialize_fp(&s[4])?,
-        s5: serialize_fp(&s[5])?,
-        s6: serialize_fp(&s[6])?,
+        s0: s[0].clone().into(),
+        s1: s[1].clone().into(),
+        s2: s[2].clone().into(),
+        s3: s[3].clone().into(),
+        s4: s[4].clone().into(),
+        s5: s[5].clone().into(),
+        s6: s[6].clone().into(),
     })
 }
 
-fn serialize_fp(value: &Fq) -> napi::bindgen_prelude::Result<Vec<u8>> {
-    let mut bytes = Vec::new();
-    value
-        .serialize_uncompressed(&mut bytes)
-        .map_err(|err| Error::new(Status::GenericFailure, format!("serialize_fp: {err}")))?;
-    Ok(bytes)
+impl From<NapiFqPlonkVerifierIndex> for DlogVerifierIndex<GAffine, OpeningProof<GAffine>> {
+    fn from(index: NapiFqPlonkVerifierIndex) -> Self {
+        of_napi(index).0
+    }
+}
+
+pub fn of_napi(
+    index: NapiFqPlonkVerifierIndex,
+) -> (
+    DlogVerifierIndex<GAffine, OpeningProof<GAffine>>,
+    Arc<SRS<GAffine>>,
+) {
+    let max_poly_size = index.max_poly_size;
+    let public_ = index.public_;
+    let prev_challenges = index.prev_challenges;
+    let log_size_of_group = index.domain.log_size_of_group;
+    let srs = &index.srs;
+    let evals = &index.evals;
+    let shifts = &index.shifts;
+
+    let (endo_q, _endo_r) = poly_commitment::ipa::endos::<GAffineOther>();
+    let domain = Domain::<Fq>::new(1 << log_size_of_group).unwrap();
+
+    let feature_flags = compute_feature_flags(&index);
```

Comment:
here instead, because it is quite long, it's okay to modularize it. Also, we might switch to macro-style later, avoiding having both fp and fq files.